test/e2e/framework/pod/wait.go

```diff
@@ -435,6 +435,19 @@ func PodsResponding(c clientset.Interface, ns, name string, wantName bool, pods
 	return wait.PollImmediate(poll, podRespondingTimeout, NewProxyResponseChecker(c, ns, label, name, wantName, pods).CheckAllResponses)
 }
 
+// WaitForNumberOfPods waits up to timeout to ensure that num pods in namespace ns are running.
+// Return the list of matching pods.
+func WaitForNumberOfPods(c clientset.Interface, ns string, num int, timeout time.Duration) (pods *v1.PodList, err error) {
```

Comment:
You can return `(pods []v1.Pod, err error)`.

Reply:
Yes, thought about this option. But I have decided to follow same style like in other function signatures at this package. There are three functions `WaitForPodsWithLabelScheduled`, `WaitForPodsWithLabel`, `WaitForPodsWithLabelRunningReady` witch returns (pods *v1.PodList, err error).